nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/clients/OAuth2Client.java

```diff
@@ -135,6 +183,94 @@ public boolean isValidWith(String clientId, String clientSecret) {
         return enabled && id.equals(clientId) && (StringUtils.isEmpty(secret) || secret.equals(clientSecret));
     }
 
+    /**
+     * Create {@link DocumentModel} from {@link OAuth2Client}.
+     *
+     * @param oAuth2Client the oAuth2Client to convert
+     * @return the corresponding {@code DocumentModel} of {@code OAuth2Client}
+     * @throws NullPointerException if oAuth2Client is {@code null}
+     * @throws NuxeoException if oAuth2Client is not valid
+     * @since 11.1
+     */
+    public static DocumentModel fromOAuth2Client(OAuth2Client oAuth2Client) {
+        return BaseSession.createEntryModel(null, OAUTH2CLIENT_SCHEMA, null, getDocumentData(oAuth2Client));
+    }
+
+    /**
+     * Update {@link DocumentModel} by {@link OAuth2Client}.
+     *
+     * @param documentModel the document model to update
+     * @param oAuth2Client the new values of document
+     * @return the updated {@code DocumentModel}
+     * @throws NullPointerException if the documentModel or oAuth2Client is {@code null}
+     * @throws NuxeoException if oAuth2Client is not valid
+     * @since 11.1
+     */
+    public static DocumentModel updateDocument(DocumentModel documentModel, OAuth2Client oAuth2Client) {
+        requireNonNull(documentModel, "documentModel model is required");
+        documentModel.setProperties(OAUTH2CLIENT_SCHEMA, OAuth2Client.getDocumentData(oAuth2Client));
+        return documentModel;
+    }
+
+    /**
+     * Get document properties from {@link OAuth2Client}.
+     *
+     * @param oAuth2Client the oAuth2Client
+     * @return the {@code Map} structure of {@code OAuth2Client}
+     * @throws NullPointerException if the oAuth2Client is {@code null}
+     * @throws NuxeoException if oAuth2Client is not valid
+     * @since 11.1
+     */
+    public static Map<String, Object> getDocumentData(OAuth2Client oAuth2Client) {
+        validate(oAuth2Client);
```

Comment:
What do you think of calling `validate` directly in `OAuth2ClientServiceImpl#create` and `OAuth2ClientServiceImpl#update` ?
It's less factorized but I think it would be clearer to explicitely validate the `OAuth2Client` parameter passed to each REST API endpoint, rather then having it hidden in the submethods.

Reply:
i see, if we do this, we can end up with saving an invalid OAuthClient, that's why i put this method in the toMap (old getDocumentData), plus the factorisation